benchmark/auto/auto_benchmark.py

```diff
@@ -0,0 +1,85 @@
+import argparse
+import json
+import os
+import random
+import pandas as pd
+from datasets import Dataset
+from typing import List, Any
+
+from rageval.models.openai import OpenAILLM
+from prompt import (SYNTHETIC_QUERY_FEW_SHOT, SYNTHETIC_QUERY_SYSTEM, SYNTHETIC_QUERY_USER, SYNTHETIC_ANSWER_SYSTEM, SYNTHETIC_ANSWER_USER)
+
+def load_corpus(corpus_dir):
+    with open(f"{corpus_dir}/corpus.json", "r", encoding="utf-8") as f:
+        docs = json.load(f)
+    df = pd.DataFrame(docs)
+    df.drop_duplicates(inplace=True)
+    dataset = Dataset.from_dict({'document':df[0].apply(lambda x: x.strip())})
+
+    with open(f"{corpus_dir}/few_shot_cases.json", "r", encoding="utf-8") as f:
+        cases = json.load(f)
+    cases = random.sample(cases, 3)
+    return dataset, cases
+
+def generate_responses(engine: OpenAILLM, user_prompts: List[List[str]], system_prompt: List[str]) -> List[str]:
+    '''Generate responses from the OpenAILLM model.'''
+    responses = engine.batch_generate(user_prompts, system_roles=system_prompt * len(user_prompts))
+    response_texts = [r.generations[0][0].text for r in responses]
+    return response_texts
+
+def generate_questions(engine: OpenAILLM, dataset: Dataset, cases) -> Dataset:
+    system_prompt = [SYNTHETIC_QUERY_SYSTEM]
+    few_shot_cases = ""
+    for i in range(len(cases)):
+        few_shot_cases += SYNTHETIC_QUERY_FEW_SHOT.format(
+            document=cases[i]["document"], question=cases[i]["Query"])
+    user_prompts = [[SYNTHETIC_QUERY_USER.format(
+        few_shot_cases=few_shot_cases, document=d['document'])] for d in dataset]
+    
+    questions = generate_responses(engine, user_prompts, system_prompt)
+    return dataset.add_column("question", questions)
+
+def generate_answers(engine: OpenAILLM, dataset: Dataset) -> Dataset:
+    system_prompt = [SYNTHETIC_ANSWER_SYSTEM]
+    user_prompts = [[SYNTHETIC_ANSWER_USER.format(
+        question=d['question'], document=d['document']) + "\n"] for d in dataset]
+    
+    answers = generate_responses(engine, user_prompts, system_prompt)
+    return dataset.add_column("answer", answers)
+
+def validate_questions(dataset: Dataset) -> Dataset:
```

Comment:
Suggest the function name to be `validate_question_with_answer`. By the way, we  can add more rules to check questions and answers.

Reply:
Fixed.